### PR TITLE
adds pagination support for environments

### PR DESCRIPTION
**Pull request by ghepting**
These changes are required in order to fetch > 25 environments

### CODE REVIEW

**Comment by rubydog** [comment]
Hi @ghepting, thanks for the PR. Is it possible to add a test case for the suggested changes?

Thanks,
Bhushan

**Comment by ghepting** [comment]
Unfortunately this repo is setup with VCR instead of more useful stubbing. I don't have access to space `facgnwwgj5fe` which the VCR cassettes are recorded from. We'll need to modify that to have 26+ environments and add another context after [spec/lib/contentful/management/environment_spec.rb#13](https://github.com/contentful/contentful-management.rb/blob/master/spec/lib/contentful/management/environment_spec.rb#L13-L20) to include `with pagination` which I can add but will be failing until we update the VCR cassette. Can you do that if I write the spec?

File: spec/lib/contentful/management/environment_spec.rb [hunk 1]
```diff
@@ -10,22 +10,62 @@
   subject { client.environments(space_id) }
 
   describe '.all' do
```
Comment (ghepting) [inline spec/lib/contentful/management/environment_spec.rb:12]: @rubydog can you please add 24 additional environments (for a total of 26) to the space: `facgnwwgj5fe` and then do the following to record new VCR fixtures for the `.all` tests with and without pagination params?

```
rm spec/fixtures/vcr_cassettes/environment/all.yml
```

```
bundle exec rspec spec/lib/contentful/management/environment_spec.rb`
```

Comment (ghepting) [reply spec/lib/contentful/management/environment_spec.rb:12]: I added temporary fixtures from one of our spaces to validate the tests, you'll just want to swap out the VCR fixtures and address the `TODO: @rubydog` I assigned you in the spec file

**Comment by rubydog** [comment]
Thanks @ghepting, I will take a look at this PR next week and write to you if I have any questions. But generally speaking, you don't need access to `facgnwwgj5fe` for VCR. Just run tests using your environment and once the VCR file is created, replace the space_id, access_token in newly created VCR file and the spec with any random value. The important part is the values should be same in yml and spec file.

**Comment by rubydog** [comment]
Hi @ghepting, did you refer to https://www.contentful.com/developers/docs/references/content-management-api/#/introduction/collection-resources-and-pagination for fetching more than 25 environments?

**Comment by ghepting** [comment]
@rubydog please do one final review. manually modified the vcr fixtures with generic data

**Comment by rubydog** [comment]
> Hi @ghepting, did you refer to https://www.contentful.com/developers/docs/references/content-management-api/#/introduction/collection-resources-and-pagination for fetching more than 25 environments?

Did you try this solution?

**Comment by ghepting** [comment]
> > Hi @ghepting, did you refer to https://www.contentful.com/developers/docs/references/content-management-api/#/introduction/collection-resources-and-pagination for fetching more than 25 environments?
> 
> Did you try this solution?

What do you mean? My implementation is _enabling_ using the limit/skip/other parameters so that I can use this API feature...?

**Comment by ghepting** [comment]
@rubydog did you have any further questions or requirements?

**Review by rubydog — APPROVED** [review]

**Comment by rubydog** [comment]
Hi @ghepting the PR looks good, can you please look at the failing tests? Once the build is green, we will merge it in master.

Cheers

**Comment by ghepting** [comment]
> Hi @ghepting the PR looks good, can you please look at the failing tests? Once the build is green, we will merge it in master.
> 
> Cheers

Ah yeah sorry I didn't notice the rubocop issue. Resolved.

**Comment by ghepting** [comment]
Let me know once this is released and i'll get back on the main rubygems source. Thanks @rubydog

**Comment by rubydog** [comment]
Hi @ghepting, it is released and available in latest version `3.6.0`. https://github.com/contentful/contentful-management.rb/blob/master/CHANGELOG.md#360

**Pull request merged**